### ISSUE TITLE
Wrong derivation for Type[T]

### ISSUE THREAD

**Issue by Fatsie** [microsoft/pylance-release#1468]
I have following minimal non-working example:
```python
from typing import Generator, List, Type, TypeVar, Union

class LI(List[int]):
    pass
class LS(List[str]):
    pass

_iter_type = TypeVar("_iter_type")
class MyList(List[Union[LI, LS]]):
    def __iter_type__(self, *, type_: Type[_iter_type]) -> Generator[_iter_type, None, None]:
        for elem in self:
            if isinstance(elem, type_):
                yield elem

l = MyList()
def same(other: Union[LI, LS]):
    for elem in l.__iter_type__(type_=other.__class__):
        for v in elem:
            print(v)
```

The problem seems to be that the derivation of `type_` argument for `__iter_type__()` on line 17 seems to be wrong; pop-up says `type_` is `Type[Type[LI] | Type[LS]]` and it should be `Type[LI] | Type[LS]`. This seems to be a regression as it did not give errors before last update of vscode.
This gives the following errors:
```JSON
{
	"resource": "/home/verhaegs/eda/code/PDKMaster/pdkmaster/test_iter.py",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"external": "https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules",
			"path": "/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "diagnostic-severity-rules"
		}
	},
	"severity": 8,
	"message": "\"Type[LI]\" is not iterable\n  \"__iter__\" method not defined",
	"source": "Pylance",
	"startLineNumber": 18,
	"startColumn": 18,
	"endLineNumber": 18,
	"endColumn": 22
}
{
	"resource": "/home/verhaegs/eda/code/PDKMaster/pdkmaster/test_iter.py",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": {
		"value": "reportGeneralTypeIssues",
		"target": {
			"$mid": 1,
			"external": "https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules",
			"path": "/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "diagnostic-severity-rules"
		}
	},
	"severity": 8,
	"message": "\"Type[LS]\" is not iterable\n  \"__iter__\" method not defined",
	"source": "Pylance",
	"startLineNumber": 18,
	"startColumn": 18,
	"endLineNumber": 18,
	"endColumn": 22
}
```

Software used:
- Pylance 2021.6.2
- vscode: latest flatpak version 1.56.2-1620837933
- OS: Ubuntu 20.04.2 LTS
- Python: anaconda 3.7.10

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release. I've added a simplified version of your code to our unit tests since it covers a case that was previously not included in our testing.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.6.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202163-23-june-2021
